cheroot/test/test_ssl.py

```diff
@@ -757,3 +799,141 @@ def test_http_over_https_error(
         'The underlying error is {underlying_error!r}'.format(**locals())
     )
     assert expected_error_text in err_text
+
+
+@pytest.mark.parametrize('adapter_type', ('builtin', 'pyopenssl'))
+@pytest.mark.parametrize(
+    'encrypted_key',
+    (True, False),
+    ids=('encrypted-key', 'unencrypted-key'),
+)
+@pytest.mark.parametrize(
+    'password_as_bytes',
+    (True, False),
+    ids=('with-bytes-password', 'with-str-password'),
+)
+# pylint: disable-next=too-many-positional-arguments
+def test_ssl_adapters_with_private_key_password(
+    http_request_timeout,
+    private_key_password,
+    tls_http_server,
+    tls_ca_certificate_pem_path,
+    tls_certificate_chain_pem_path,
+    tls_certificate_passwd_private_key_pem_path,
+    tls_certificate_private_key_pem_path,
+    adapter_type,
+    encrypted_key,
+    password_as_bytes,
+):
+    """Check server start and response when using ssl adapter with password-protected private key."""
+    httpserver = HTTPServer(
+        bind_addr=(ANY_INTERFACE_IPV4, EPHEMERAL_PORT),
+        gateway=Gateway,
+    )
+    key_file = (
+        tls_certificate_passwd_private_key_pem_path
+        if encrypted_key
+        else tls_certificate_private_key_pem_path
+    )
+    key_pass = (
+        private_key_password.encode('utf-8')
+        if password_as_bytes
+        else private_key_password
+    )
+
+    tls_adapter_cls = get_ssl_adapter_class(name=adapter_type)
+    tls_adapter = tls_adapter_cls(
+        certificate=tls_certificate_chain_pem_path,
+        private_key=key_file,
+        private_key_password=key_pass,
+    )
+
+    httpserver.ssl_adapter = tls_adapter
+
+    httpserver.prepare()
+    assert httpserver.ready
+    httpserver.stop()
+
+    interface, _host, port = _get_conn_data(
+        tls_http_server(httpserver.bind_addr, tls_adapter).bind_addr,
```

Comment:
Why aren't you using the same server that was used to check TLS adapter loading the certs? This feels like having two different tests under the same roof. My idea was that the same server that loaded the certificates would be used for HTTPS requests and this working would be the final assertion that all's good.